parser/src/aspects/literal.rs

```diff
@@ -71,34 +90,33 @@ impl<'a> LiteralAspect<'a> for Parser<'a> {
                 }
             };
         }
-        Ok(Expr::Literal(Literal {
+        Ok(Literal {
             lexeme,
             parsed: LiteralValue::String(value),
-        }))
+        })
     }
 
-    fn templated_string_literal(&mut self) -> ParseResult<Expr<'a>> {
-        let start = self
-            .cursor
-            .force(of_type(TokenType::DoubleQuote), "Expected quote.")?;
+    fn templated_string_literal(&mut self) -> ParseResult<TemplateString<'a>> {
+        self.cursor.force(of_type(DoubleQuote), "Expected quote.")?;
+
         let mut lexeme = self.cursor.peek().value;
         let mut literal_value = String::new();
         let mut parts = Vec::new();
         loop {
             if self.cursor.is_at_end() {
                 return self.expected(
                     "Unterminated string literal.",
-                    ParseErrorKind::Unpaired(self.cursor.relative_pos(&start)),
+                    ParseErrorKind::Unpaired(self.cursor.relative_pos(lexeme)),
                 );
```

Comment:
The unpaired error should point at the unpaired token, i.e. the quote. If the lexeme is used, the start annotation is wrong.
```
  × Unterminated string literal.
   ╭─[stdin:1:1]
 1 │ "aaa
   ·  ─┬─┬
   ·   │ ╰── Here
   ·   ╰── Start
   ╰────
```
The `Unexcepted` error would be more appropriate if you want to annotate the whole lexeme:
```rs
return self.expected_with(
    "Unterminated string literal.",
    lexeme,
    ParseErrorKind::Unexpected
);
```
This also applies to the `string_literal` method.

Reply:
prefer it with the 'start' cursor, i'll fix that